aiida_common_workflows/workflows/relax/generator.py

```diff
@@ -93,33 +98,33 @@ def process_class(self):
     @abstractmethod
     def get_builder(
         self,
-        structure,
-        calc_engines,
-        protocol,
-        relaxation_type,
-        threshold_forces=None,
-        threshold_stress=None,
+        structure: StructureData,
+        calc_engines: Dict[str, Any],
+        protocol: str,
+        *,
+        relax_type: RelaxType = RelaxType.ATOMS,
+        electronic_type: ElectronicType = ElectronicType.METAL,
+        spin_type: SpinType = SpinType.NONE,
+        magnetization_per_site: List[float] = None,
```

Comment:
Didn't we agree to leave it as a list where we can pass more than just floats? We did not discuss extensively the support for non-collinear situations, but I think it will be very soon settled.

Reply:
Since for now we only support `NONE` and `COLLINEAR` and for the latter we only need a list of floats, I though to keep it as this. Once we add official support for non collinear and spin orbit we can update the typing. Prefer to have a type now if we haven't settled on the format of the other forms of spin yet

Reply:
Ok.